reputation-bar/reputation-bar.lua

```diff
@@ -16,39 +16,62 @@
 ---	See the License for the specific language governing permissions and
 ---	limitations under the License.
 ----------------------------------------------------------------------------------
+---
+
+local Ellyb = Ellyb(...);
 
 ---@type StatusBar
 local StorylineReputationBar = Storyline_NPCFriendshipStatusBar;
 
--- API
-local GetFriendshipReputation = GetFriendshipReputation;
-
-
 local customReputationColors = {
 	[1391545]   = CreateColor(0.227,0.203,0.745, 1), -- Arcane thirst of the Nightfallen
 	["DEFAULT"] = CreateColor(0.709,0.396,0.031, 1),
 }
 local DEFAULT_ICON = [[Interface\Common\friendship-heart]];
 
-hooksecurefunc("NPCFriendshipStatusBar_Update", function(_, factionID --[[ = nil ]])
-	local id, rep, maxRep, name, text, texture, reaction, threshold, nextThreshold = GetFriendshipReputation(factionID);
-	StorylineReputationBar.friendshipFactionID = id;
-	if ( id and id > 0 ) then
-		if ( not nextThreshold ) then
-			threshold, nextThreshold, rep = 0, 1, 1;
+local function OnReputationBarLoad(self)
+	self:SetStatusBarTexture(1, 1, 1, "BORDER", -1);
```

Comment:
I don't believe this is valid anymore in 10.0; they simplified this API to only accept a single "asset" argument as a string if the autogenerated API documentation is to be believed. I suspect the request as such for the "BORDER" draw layer is ignored, along with the sublevel.

I'd suggest the clearer alternative of manually creating a texture object and assigning it via the SetStatusBarTexture method. The API documentation lies a bit here and does accept this still it seems:

```lua
local texture = self:CreateTexture(nil, "BORDER", nil, -1);
texture:SetGradient("VERTICAL", ...);
self:SetStatusBarTexture(texture);
```